### DISCUSSION TITLE
Question: how to add different contributor types in templates?

### DISCUSSION THREAD

**Discussion by netgrims** [windingwind/zotero-better-notes#865] [Q&A]
I've been trying to add different types of contributors to my template, for example authors, editors, book editors, contributors, etc. I have tried all of the below:

```
${topItem.getField("editor")}

${topItem.getField("author")}

${topItem.getField("creator")}

${topItem.getField("seriesEditor")}

${topItem.getField("collection-editor")}

${topItem.getField('AuthorL')}

${topItem.getField('authors')}

${topItem.getCreators()}

${({
const creators = topItem.getCreators();
creators[0].lastName;
})}
```
None of these commands work. I tried to find the answer by searching this repo, but I couldn't find it. If anyone knows how to do this, feel free to help out.

**Comment by windingwind** [comment]
not all fields are available in all types of items. see https://[aurimasv.github.io/z2csl/typeMap.xml](https://aurimasv.github.io/z2csl/typeMap.xml)

I think `${topItem.getCreators()}` won't work, since we expect a string but `getCreators` returns an object/array?

and 

```js
${({
const creators = topItem.getCreators();
creators[0].lastName;
})}
```

It definitely won't work too, since it's not a valid JS expression.

**Reply by windingwind** [reply]
Please read the doc: https://github.com/windingwind/zotero-better-notes/blob/master/docs/about-note-template.md#script-syntax

**Reply by netgrims** [reply]
Yes, I have looked at both links. What I'm mostly trying to get at is that the field identifiers for "author" and such don't work. When using `${topItem.getCreators()}`, the array contains all contributors, but there seems to be no way to differentiate in that array between authors, editors, book editors, contrbutors, etc. It's just an array that contains all contributors, in the order that they were entered into Zotero.

I know there is `${topItem.getField('firstCreator')}`, but is there any equivalent for other kinds of contributors that allows users to differentiate between the different possible types of contributors? In the list of Zotero fields (https://aurimasv.github.io/z2csl/typeMap.xml), `author` is a valid type for almost every item type, so I don't entirely understand why it doesn't work.

**Reply by windingwind** [reply]
We can only use `Zotero field`, not the `CSL field`. It also depends on your item in Zotero, which might not have the corresponding data when importing from the web/file.

**Reply by netgrims** [reply]
Yeah, I've been looking at the zotero fields. Many of them do include `author (author)` or `editor` (again, on this website: https://aurimasv.github.io/z2csl/typeMap.xml). Yet even when I try them on items that definitely have authors and/or editors, I can't get them to work, for example with `${topItem.getField("editor")}`
